src/chunker/chunk_g1_points.rs

```diff
@@ -0,0 +1,165 @@
+use super::elements::DataType::{FqData,G1PointData};
+use super::elements::{FqType, G1PointType};
+
+use crate::bn254::fp254impl::Fp254Impl;
+use crate::bn254::fq::Fq;
+use crate::chunker::assigner::*;
+use crate::chunker::elements::ElementTrait;
+use crate::chunker::segment::Segment;
+use crate::treepp::script;
+
+use ark_bn254::Bn254;
+use ark_ff::Field;
+use ark_groth16::{Proof, VerifyingKey};
+use core::ops::Neg;
+
+pub fn g1_points<T: BCAssigner>(
+    assigner: &mut T,
+    g1p: G1PointType,
+    g1a: ark_bn254::G1Affine,
+    proof: &Proof<Bn254>,
+    vk: &VerifyingKey<Bn254>,
+) -> (Vec<Segment>, Vec<FqType>) {
+    let mut segments = vec![];
+
+    let ( p2, p3, p4) = (proof.c, vk.alpha_g1, proof.a);
+    let mut g2p = G1PointType::new(assigner, &format!("{}", "F_p2_init"));
+    g2p.fill_with_data(G1PointData(p2));
+    let mut g3p = G1PointType::new(assigner, &format!("{}", "F_p3_init"));
+    g3p.fill_with_data(G1PointData(p3));
+    let mut g4p = G1PointType::new(assigner, &format!("{}", "F_p4_init"));
+    g4p.fill_with_data(G1PointData(p4));
+
+    let (s1, a1, b1) = p(assigner,"F_p1_im".to_owned(), g1p, g1a);
+    let (s2, a2, b2) = p(assigner,"F_p2_im".to_owned(), g2p, p2);
+    let (s3, a3, b3) = p(assigner,"F_p3_im".to_owned(), g3p, p3);
+    let (s4, a4, b4) = p(assigner,"F_p4_im".to_owned(), g4p, p4);
+
+    segments.extend(s1);
+    segments.extend(s2);
+    segments.extend(s3);
+    segments.extend(s4);
+
+    let im_var_p = vec![a1, b1, a2, b2, a3, b3, a4, b4];
+    (segments, im_var_p)
+}
+
+fn p<T: BCAssigner>(
+    assigner: &mut T,
+    prefix:String,
+    g1p: G1PointType,
+    g1a: ark_bn254::G1Affine,
+) -> (Vec<Segment>, FqType, FqType) {
+    let mut segments = vec![];
+
+    let p1 = g1a;
+    let (hinted_script1, hint1) = Fq::hinted_inv(p1.y);
+    let (hinted_script2, hint2) = Fq::hinted_mul(1, p1.y.inverse().unwrap(), 0, p1.x.neg());
+    let script_lines_p1 = [
+        hinted_script1, // Fq::inv(),
+        Fq::copy(0),
+        Fq::roll(2),
+        Fq::neg(0),
+        hinted_script2, // Fq::mul()
+        Fq::roll(1),
+    ];
+    let mut script_p1 = script! {};
+    for script_line in script_lines_p1 {
+        script_p1 = script_p1.push_script(script_line.compile());
+    }
```

Comment:
I think this is because of the style that we introduced. When we wrote with the macro a big script that calls other functions, it got bigger in RAM, failed to run, and we couldnt figure out the reason. However, when we instead created a list of script lines, and compiled them, as in the code above, the problem was gone.

Reply:
fixed